pyvista/plotting/renderer.py

```diff
@@ -2369,3 +2375,207 @@ def height(self):
+
+        origin : list, optional
+            If used, specifies the x and y position of the lower left corner
+            of the legend.
+
+        face : str, optional
+            Face shape of legend face. Accepted options:
+
+            * ``'triangle'``
+            * ``'circle'``
+            * ``'rectangle'``
+            * ``None``
+
+            Default is ``'triangle'``.  Passing ``None`` removes the
+            legend face.
+
+        Returns
+        -------
+        vtk.vtkLegendBoxActor
+            Actor for the legend.
+
+        Examples
+        --------
+        Create a legend by labeling the meshes when using ``add_mesh``
+
+        >>> import pyvista
+        >>> from pyvista import examples
+        >>> sphere = pyvista.Sphere(center=(0, 0, 1))
+        >>> cube = pyvista.Cube()
+        >>> plotter = pyvista.Plotter()
+        >>> _ = plotter.add_mesh(sphere, 'grey', smooth_shading=True, label='Sphere')
+        >>> _ = plotter.add_mesh(cube, 'r', label='Cube')
+        >>> _ = plotter.add_legend(bcolor='w', face=None)
+        >>> plotter.show()
+
+        Alternatively provide labels in the plotter.
+
+        >>> plotter = pyvista.Plotter()
+        >>> _ = plotter.add_mesh(sphere, 'grey', smooth_shading=True)
+        >>> _ = plotter.add_mesh(cube, 'r')
+        >>> legend_entries = []
+        >>> legend_entries.append(['My Mesh', 'w'])
+        >>> legend_entries.append(['My Other Mesh', 'k'])
+        >>> _ = plotter.add_legend(legend_entries)
+        >>> plotter.show()
+
+        """
+        if self.legend is not None:
+            self.remove_legend()
+        self._legend = _vtk.vtkLegendBoxActor()
+
+        if labels is None:
+            # use existing labels
+            if not self._labels:
+                raise ValueError('No labels input.\n\n'
+                                 'Add labels to individual items when adding them to'
+                                 'the plotting object with the "label=" parameter.  '
+                                 'or enter them as the "labels" parameter.')
+
+            self._legend.SetNumberOfEntries(len(self._labels))
+            for i, (vtk_object, text, color) in enumerate(self._labels.values()):
+
+                if face is None:
+                    # dummy vtk object
+                    vtk_object = pyvista.PolyData([0, 0, 0])
+
+                self._legend.SetEntry(i, vtk_object, text, parse_color(color))
+
+        else:
+            self._legend.SetNumberOfEntries(len(labels))
+
+            if face is None:
+                legendface = pyvista.PolyData([0, 0, 0])
+            elif face in ['-', 'line']:
+                legendface = _line_for_legend()
+            elif face == "triangle":
+                legendface = pyvista.Triangle()
+            elif face == "circle":
+                legendface = pyvista.Circle()
+            elif face == "rectangle":
+                legendface = pyvista.Rectangle()
```

Comment:
How about following the [markers in matplotlib](https://matplotlib.org/stable/api/markers_api.html) ?
```suggestion
            elif face in ["-", "line"]:
                legendface = _line_for_legend()
            elif face in ["^", "triangle"]:
                legendface = pyvista.Triangle()
            elif face in ["o", "circle"]:
                legendface = pyvista.Circle()
            elif face in ["r", "rectangle"]:
                legendface = pyvista.Rectangle()
```

Reply:
That's way better!